megatron/model/transformer.py

```diff
@@ -723,24 +792,33 @@ def forward(self, x, attention_mask, layer_past=None):
                 attention_output, presents = attention_output
                 self.layer_past = presents
             with torch.enable_grad():
-                attention_output = bias_dropout_fn(
-                    attention_output,
-                    bias=attention_bias.expand_as(residual),
-                    residual=residual,
-                    prob=self.hidden_dropout,
-                )
+                if attention_bias is not None:
```

Comment:
This code is a bit opaque to me. Is this check supposed to be for whether or not we have bias weights? I think we need to have some comments explaining it.